lib/manager/pip_requirements/extract.js

```diff
@@ -16,6 +16,13 @@ module.exports = {
 function extractDependencies(content) {
   logger.debug('pip_requirements.extractDependencies()');
 
+  let registryUrls;
+  const [firstLine] = content.split('\n');
```

Comment:
The `--index-url` option can technically come on any line before the first package is specified.

The following reference shows that options are first in the file, and that options may be given in any order - https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format

Reply:
I've modified it to look on *every* line, as that should catch all valid configurations without being too complex.